[ Hu HongRui ] iP

muller317 opened the pull request:
> “Your mind is for having ideas, not holding them.” – David Allen

MullerPro frees your mind from having to remember things you need to do. It's
- text-based
- easy to learn
- SUPER FAST to use

All you need to do is,
1. Download it from here.
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you!

And it's FREE!
Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
`public static void main(String[] args) {
        new Muller("data/muller.txt").run();
    }`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

muller317 committed:
deleted duke

muller317 committed:
Level-0

muller317 committed:
Level-1

muller317 committed:
level-2

muller317 committed:
Level-3

muller317 committed:
Level 4

muller317 committed:
close scanner

muller317 committed:
A-TextUiTesting

muller317 committed:
Level-5

muller317 committed:
Level-6

muller317 committed:
bug

muller317 committed:
bug

muller317 committed:
reformat try and catch block

muller317 committed:
Level-7

muller317 committed:
Merge branch 'branch-Level-7'

muller317 committed:
Level 8. Dates and Times

muller317 committed:
Merge branch 'branch-Level-8'

muller317 committed:
refactor: restructure Muller chatbot using OOP principles

- Extracted core functionality into separate classes to improve modularity:
  - `Ui`: Handles all user interactions (input/output).
  - `Storage`: Manages loading and saving tasks to/from a file.
  - `Parser`: Parses user commands and creates corresponding Command objects.
  - `TaskList`: Encapsulates the task list and provides methods for managing tasks.
  - `Command` classes (`AddCommand`, `DeleteCommand`, `MarkCommand`, etc.): Implement specific actions and inherit from an abstract `Command` class.

muller317 committed:
Merge branch 'Branch-A-MoreOOP'

muller317 committed:
refactor: organize classes into appropriate packages

- Moved classes into specific packages to improve modularity and maintainability:
  - `muller.ui`: Contains classes related to user interaction.
  - `muller.storage`: Manages loading and saving tasks from/to files.
  - `muller.parser`: Handles command parsing.
  - `muller.task`: Encapsulates task-related classes, including `Task` and `TaskList`.
  - `muller.command`: Contains command execution classes such as `AddCommand`, `DeleteCommand`, `MarkCommand`, etc.

- Updated import statements and compilation paths to reflect the new package structure.

This reorganization enhances the clarity of the codebase, making it easier to navigate.

muller317 committed:
Merge branch 'Branch-A-Packages'

muller317 committed:
Merge branch 'add-gradle-support'

muller317 committed:
build: add Gradle support to automate project builds

- Integrated Gradle into the project to automate build tasks, following the SE-EDU Gradle tutorial.
- Configured the `build.gradle` file to handle project compilation, execution, and dependency management.
- Added tasks to build and run the Muller application using Gradle.
- Updated the project structure to be compatible with Gradle's build system.
- This setup lays the foundation for further automation, including running JUnit tests and other tasks as the project progresses.

This change streamlines the build process and enhances project maintainability.

muller317 committed:
Merge branch 'Branch-A-Gradle'

muller317 committed:
test: add JUnit tests for Task and TaskList classes

- Added JUnit tests for the Task class, covering methods like setDate().
- Added JUnit tests for the TaskList class, covering methods like addTask() and deleteTask().
- Configured Gradle to include JUnit for running tests.
- Ensured tests are placed in the correct directory structure under src/test/java/muller.

These tests help ensure the reliability of the core task management functionalities.

muller317 committed:
Merge branch 'Branch-A-JUnit'

muller317 committed:
docs: add JavaDoc comments to classes and methods

- Added JavaDoc comments to all non-private classes and methods across the project.
- Provided detailed descriptions of the purpose, parameters, and return values for each method.
- Included comments for non-trivial private methods to improve code readability and maintainability.
- Ensured that the JavaDoc comments adhere to standard conventions, making the codebase easier to understand for future developers.

This documentation enhances the clarity of the code and facilitates better collaboration and maintenance.

muller317 committed:
Merge branch 'Branch-A-JavaDoc'

muller317 committed:
feat: add find command to search tasks by keyword

- Implemented FindCommand to allow users to search for tasks by a keyword in the task description.
- Updated Parser to recognize the 'find' command.
- Added findTasksByKeyword method to TaskList to search tasks based on the keyword.
- Enhanced Ui to display the list of tasks that match the search criteria.

This feature allows users to quickly locate specific tasks by searching for keywords in their descriptions.

muller317 committed:
Merge branch 'Branch-Level-9'

IzN432 commented on src/main/java/muller/Muller.java:
```diff
@@ -0,0 +1,63 @@
+                String fullCommand = ui.readCommand();
+                ui.showLine(); // show the divider line ("_______")
+                Parser p = new Parser();
+                Command c = p.parse(fullCommand);
```
Would it be better if you used the full words ```parser``` and ```command``` as variable names instead?

IzN432 commented on src/main/java/muller/command/OnCommand.java:
```diff
@@ -0,0 +1,51 @@
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        ui.showLine();
+        System.out.println("Tasks on " + date.format(Task.OUTPUT_DATE_FORMATTER) + ":");
+        boolean found = false;
```
Perhaps consider naming it ```isFound``` instead to make it clearer that it is a boolean?

IzN432 commented on src/main/java/muller/storage/Storage.java:
```diff
@@ -0,0 +1,110 @@
+     * @throws MullerException If there is an issue loading the tasks.
+     */
+    public ArrayList<Task> load() throws MullerException {
+        ArrayList<Task> list = new ArrayList<>();
```
Perhaps rename this to ```tasks``` to make it a bit clearer?

IzN432 commented on src/main/java/muller/task/Task.java:
```diff
@@ -0,0 +1,149 @@
+     *
+     * @return The icon representing the task's completion status.
+     */
+    public String doneIcon() {
```
Maybe use a verb for method name such as getDoneIcon?

IzN432 commented on src/main/java/muller/ui/Ui.java:
```diff
@@ -0,0 +1,91 @@
+ * Handles all user interactions, including displaying messages and reading user input.
+ */
+public class Ui {
+    private Scanner sc;
```
Perhaps you could just rename this to scanner to be clearer?

IzN432 reviewed (commented):
LGTM! Just some tiny nits about some of your variable and method names.

DKHB0207 commented on src/main/java/muller/Muller.java:
```diff
@@ -0,0 +1,63 @@
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine(); // show the divider line ("_______")
+                Parser p = new Parser();
```
Is there a reason why the Parser instantiation is within the while loop? Perhaps it could be instantiated outside the while loop to prevent unnecessary object creation.

DKHB0207 commented on src/main/java/muller/command/AddCommand.java:
```diff
@@ -0,0 +1,121 @@
+ * Command to add a task (Todo, Deadline, or Event) to the task list.
+ */
+public class AddCommand extends Command {
+    private String[] inputs;
```
Perhaps the use of more descriptive names for method parameters would be better? eg. commandInputs or taskType

DKHB0207 commented on src/main/java/muller/command/AddCommand.java:
```diff
@@ -0,0 +1,121 @@
+
+        tasks.addTask(task);
+        ui.showLine();
+        System.out.println("Got it. I've added this task:");
```
Would it be better to specify a method for printing statements in the UI class instead?

DKHB0207 commented on src/main/java/muller/storage/Storage.java:
```diff
@@ -0,0 +1,110 @@
+     * @return An ArrayList of Task objects loaded from the file.
+     * @throws MullerException If there is an issue loading the tasks.
+     */
+    public ArrayList<Task> load() throws MullerException {
```
Perhaps a more specific method name such as loadTasks and saveTasks will be better?

DKHB0207 commented on src/main/java/muller/task/TaskList.java:
```diff
@@ -0,0 +1,110 @@
+     *
+     * @return The size of the task list.
+     */
+    public int size() {
```
Perhaps using a name like getSize would be more appropriate as methods should be phrased as verbs?

DKHB0207 commented on src/main/java/muller/task/Task.java:
```diff
@@ -0,0 +1,149 @@
+     *
+     * @return The string representation of the task for saving.
+     */
+    public String toFileString() {
```
This could be changed to a more descriptive method name such as convertToFileString.

DKHB0207 reviewed (commented):
Hey Hong Rui! LGTM. Just some nits from me, generally your code looks good and your functions look robust. However, most of your printing statements are in the execute methods, which may be better to be specified in the Ui class instead. I've only commented on the first occurrence of it to reduce the amount of repeated comments, but do take a look at all of the Command files.

Overall, the code is well done and coding standard is generally adhered to. Good job!

muller317 committed:
feat: implemented multiple name changes to variables and function names

-changed the variable name from c and p to command and parser respectively under run() in Muller.java.
-changed the variable name inputs and type to commandInputs and taskType respectively under AddCommand.java.
-delegated the print method under Ui.java by adding a new method showTaskAdded.
-changed the variable name found to inFound under execute() in OnCommand.java
-changed the function name doneIcon() to getDoneIcon() in Task.java
-changed the function name toFileString() to converttoFileString() in Task.java
-changed the function name size() to getSize() in TaskList.java

Overall these changes increased the readability of the code

muller317 committed:
feat: implemented multiple name changes to variables and function names

-changed the variable name from c and p to command and parser respectively under run() in Muller.java.
-changed the variable name inputs and type to commandInputs and taskType respectively under AddCommand.java.
-delegated the print method under Ui.java by adding a new method showTaskAdded.
-changed the variable name found to inFound under execute() in OnCommand.java
-changed the function name doneIcon() to getDoneIcon() in Task.java
-changed the function name toFileString() to converttoFileString() in Task.java
-changed the function name size() to getSize() in TaskList.java

Overall these changes increased the readability of the code

muller317 committed:
Revert "feat: implemented multiple name changes to variables and function names"

This reverts commit 65b0fcee0337869c087c7ed67d081bd96c040f1d.

muller317 committed:
Merge branch 'Branch-A-Checkstyle'

muller317 committed:
feat: integrate JavaFX for GUI in the chatbot

- Implemented a graphical user interface (GUI) for the chatbot using JavaFX.
- Created a basic layout for the chatbot with a text input field, a display area for messages, and a submit button.
- Added event handling for user input to process commands and display chatbot responses in the GUI.
- Refactored existing chatbot logic to work with the JavaFX.

This integration improves the user experience by replacing the text-based interface with a more intuitive graphical interface.

muller317 committed:
Merge branch 'Branch-A-Level-10'

muller317 committed:
Add assertions to ensure consistency

Several methods across the Parser, Task, Storage, and Ui classes currently rely on assumptions about valid input or object state that are not explicitly enforced.

Assertions help catch programmer errors during development by ensuring that these assumptions are always met.

This commit introduces assertions to:

    Validate non-null parameters in tasks, command, UI methods
    Confirm task names are never null or empty
    Check that internal states (e.g., task lists, input strings) are valid

Using assertions improves the reliability of the codebase by ensuring internal consistency and preventing unnoticed programmer mistakes during development. These assertions are disabled in production to avoid performance overhead.

muller317 committed:
Merge pull request #2 from muller317/branch-A-Assertions

Add assertions to ensure consistency

muller317 committed:
Refactor command classes to reduce code duplication

The DeleteCommand, MarkCommand, and UnmarkCommand classes have duplicate code for validating task indices and checking numeric input.

Duplicated validation logic can introduce inconsistencies and makes maintaining the codebase harder, as changes to validation logic need to be repeated across multiple files.

This commit extracts the common validation logic into a utility class CommandUtils to:

    Centralize index and numeric validation
    Improve code maintainability
    Reduce duplication

By doing this, we ensure the validation logic is consistent across all relevant commands, making future changes easier to implement and reducing the risk of bugs.

muller317 committed:
Merge pull request #3 from muller317/branch-A-CodeQuality

Refactor command classes to reduce code duplication

muller317 committed:
Update README.md

muller317 committed:
Add a full GUI window for Muller chatbot.

muller317 committed:
Enable the program to exit after exit command

muller317 committed:
Add a time delay before the application exit

muller317 committed:
Merge branch 'master' of https://github.com/muller317/ip

muller317 committed:
Fix the logic for checking valid inputs for various commands

muller317 committed:
Create different task type subclasses to align towards the OOP standard

muller317 committed:
Enable the chatbot to give reminders about tasks

muller317 committed:
Merge pull request Branch-B-Reminders

Enable the chatbot to give reminders about tasks

muller317 committed:
Update README.md

muller317 committed:
Merge pull request Branch-A-UserGuide

Update README.md

muller317 committed:
updated Readme
